zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/CreateProcessInstanceWithSuspensionInstructionTest.java

```diff
@@ -0,0 +1,560 @@
+            .create();
+
+    // then
+    final var result =
+        RecordingExporter.userTaskRecords(UserTaskIntent.CANCELED)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+
+    assertThat(result.getValue().getElementId()).isEqualTo("userTask");
+  }
+
+  @Test
+  public void shouldResolveIncidentsWhenProcessInstanceIsSuspended() {
+    // given
+    final String processId = "process";
+    final String elementToSuspend = "element";
+    ENGINE
+        .deployment()
+        .withXmlResource(
+            Bpmn.createExecutableProcess(processId)
+                .startEvent()
+                .parallelGateway()
+                // invalid timer expression will trigger an incident
+                .intermediateCatchEvent(
+                    "timer", e -> e.timerWithDurationExpression("not.a.valid.expr"))
+                .endEvent()
+                .moveToLastGateway()
+                .manualTask(elementToSuspend)
+                .endEvent()
+                .done())
+        .deploy();
+
+    // when
+    final long processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(processId)
+            .withRuntimeSuspendInstruction(elementToSuspend)
+            .create();
+
+    // then
+    final var result =
+        RecordingExporter.incidentRecords(IncidentIntent.RESOLVED)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+
+    assertThat(result.getValue().getElementId()).isEqualTo("timer");
+  }
+
+  @Test
+  public void shouldBePossibleToCancelSuspendedProcessInstance() {
+    final String processId = "process";
+    final String elementToSuspend = "element";
+    ENGINE
+        .deployment()
+        .withXmlResource(
+            Bpmn.createExecutableProcess(processId)
+                .startEvent()
+                .parallelGateway()
+                .task("task1")
+                .task("task2")
+                .task("task3")
+                .task("task4")
+                .endEvent()
+                .moveToLastGateway()
+                .manualTask(elementToSuspend)
+                .endEvent()
+                .done())
+        .deploy();
+
+    // when
+    final long processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(processId)
+            .withRuntimeSuspendInstruction(elementToSuspend)
+            .create();
+
+    // then
+    final var result =
+        RecordingExporter.processInstanceRecords()
+            .onlyEvents()
+            .withProcessInstanceKey(processInstanceKey)
+            .limit(processId, ProcessInstanceIntent.ELEMENT_SUSPENDED)
+            .filter(record -> record.getValue().getElementId().equals(processId));
+
+    assertThat(result)
+        .extracting(Record::getIntent, record -> record.getValue().getElementId())
+        .contains(Tuple.tuple(ProcessInstanceIntent.ELEMENT_SUSPENDED, processId));
```

Comment:
🔧 this can be simplified, because it's just looking for a single entry

```java
    final var result =
        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_SUSPENDED)
            .withProcessInstanceKey(processInstanceKey)
            .getFirst();

    Assertions.assertThat(result.getValue()).hasElementId(processId);
```